libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/DrillTargets/AttributeHierarchyListItem.tsx

```diff
@@ -0,0 +1,75 @@
+// (C) 2023 GoodData Corporation
+
+import React from "react";
+import cx from "classnames";
+
+import { stringUtils } from "@gooddata/util";
+import { ShortenedText } from "@gooddata/sdk-ui-kit";
+
+import { areObjRefsEqual, ICatalogAttributeHierarchy } from "@gooddata/sdk-model";
+import {
+    AttributeHierarchyDetailBubble,
+    AttributeHierarchyDetailPanel,
+    IAttributeHierarchyDetailItem,
+} from "@gooddata/sdk-ui-ext";
+import {
+    selectCatalogAttributes,
+    selectCatalogDateAttributes,
+    useDashboardSelector,
+} from "../../../../../model/index.js";
+
+/**
+ * @internal
+ */
+export interface IAttributeHierarchyListItemProps {
+    item: ICatalogAttributeHierarchy;
+    onClick: () => void;
+    isSelected?: boolean;
+}
+export const AttributeHierarchyListItem: React.FC<IAttributeHierarchyListItemProps> = (props) => {
+    const { onClick, item } = props;
+    const catalogAttributes = useDashboardSelector(selectCatalogAttributes);
+    const catalogDateAttributes = useDashboardSelector(selectCatalogDateAttributes);
+
+    const hierarchyListItemClassname = cx(
+        "attribute-hierarchy-list-item s-attribute-hierarchy-list-item",
+        `s-${stringUtils.simplifyText(item.attributeHierarchy.title)}`,
+    );
+    const attributeDetailItems = item.attributeHierarchy.attributes
```

Comment:
Please extract this mapper to a function outside of this component to make it more readable.

Reply:
Done